plugins/tiddlywiki/markdown/EditorToolbar/file-import.tid

```diff
@@ -0,0 +1,51 @@
+title: $:/plugins/tiddlywiki/markdown/EditorToolbar/file-import
+tags: $:/tags/EditorTools
+condition: [<targetTiddler>type[text/x-markdown]] [<targetTiddler>type[text/markdown]]
+
+\procedure lingo-base() $:/language/Import/
+
+\procedure closePopupActions()
+<$action-deletetiddler $filter="[title<importState>] [title<importTitle>]"/>
+\end
+
+\procedure replacement-text-image() ![](<#$title$>)
+
+\procedure replacement-text-file() [](<#$title$>)
+
+\procedure replacement-regexp() [()<>\\]
+
+\procedure postImportActions()
+\whitespace trim
```

Comment:
procedures inherit `\whitespace trim` from the tiddler. So you only need to set it once. Except you need it on a per procedure level.

Reply:
You also should use indentation for `<$let, <$action` and so on, to increase readability. If a procedure body is only 3 or 4 lines indentation may be skipped. `\whitespace trim` and `<$list` **can stay as they are**